src/SourceBuild/tarball/content/scripts/bootstrap/buildBootstrapPreviouslySB.csproj

```diff
@@ -0,0 +1,61 @@
+    <BaseOutputPath>$(MSBuildProjectDirectory)/artifacts/</BaseOutputPath>
+    <RestorePackagesPath>$(MSBuildProjectDirectory)/artifacts/restoredPkgs/</RestorePackagesPath>
+    <UnpackedTarPath>$(MSBuildProjectDirectory)/artifacts/unpacked/</UnpackedTarPath>
+    <NewTarballName>$(ArchiveDir)Private.SourceBuilt.Artifacts.Bootstrap.tar.gz</NewTarballName>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!-- These packages will be replaced with ms-built packages downloaded from official package feeds-->
+    <PackageReference Include="Microsoft.NET.HostModel" Version="$(MicrosoftNETHostModelVersion)" />
+    <PackageReference Include="Microsoft.ILVerification" Version="$(MicrosoftILVerificationVersion)" />
+    <PackageReference Include="Microsoft.NET.Sdk.IL" Version="$(MicrosoftNETSdkILVersion)" />
+    <PackageReference Include="Microsoft.NETCore.ILAsm" Version="$(MicrosoftNETCoreILAsmVersion)" />
+    <PackageReference Include="Microsoft.NETCore.ILDAsm" Version="$(MicrosoftNETCoreILDAsmVersion)" />
+    <PackageReference Include="Microsoft.NETCore.TestHost" Version="$(MicrosoftNETCoreTestHostVersion)" />
+    <PackageReference Include="runtime.linux-x64.Microsoft.NETCore.ILAsm" Version="$(RuntimeLinuxX64MicrosoftNETCoreILAsmVersion)" />
+    <PackageReference Include="runtime.linux-x64.Microsoft.NETCore.ILDAsm" Version="$(RuntimeLinuxX64MicrosoftNETCoreILDAsmVersion)" />
+    <PackageReference Include="runtime.linux-x64.Microsoft.NETCore.TestHost" Version="$(RuntimeLinuxX64MicrosoftNETCoreTestHostVersion)" />
+    <PackageReference Include="runtime.linux-x64.runtime.native.System.IO.Ports" Version="$(RuntimeLinuxX64RuntimeNativeSystemIOPortsVersion)" />
+  </ItemGroup>
+
+  <Target Name="BuildBoostrapPreviouslySourceBuilt" AfterTargets="Restore">
+    <ItemGroup>
+      <RestoredNupkgs Include="$(RestorePackagesPath)**/*.nupkg" />
+      <PrevSBArchive Include="$(ArchiveDir)Private.SourceBuilt.Artifacts.*.tar.gz" />
+    </ItemGroup>
+
+    <!-- Copy restored nupkg files to root of restored packages dir so they're all in one place-->
+    <Copy SourceFiles="@(RestoredNupkgs)" DestinationFolder="$(RestorePackagesPath)" />
+
+    <!-- Check to ensure there is only one previously source-built archive -->
+    <Error Text="Multiple Private.SourceBuilt.Artifacts.*.tar.gz tarballs exists at $(ArchiveDir).  Expecting only one."
+        Condition="'@(PrevSBArchive->Count())' != '1'" />
+
+    <!-- Unpack existing archive -->
+    <Message Text="  Unpacking existing tarball from %(PrevSBArchive.Identity)" Importance="High" />
+    <MakeDir Directories="$(UnpackedTarPath)" />
+    <Exec Command="tar -xzf %(PrevSBArchive.Identity) -C $(UnpackedTarPath)" />
+
+    <!-- Rename existing archive with backup file name -->
+    <Message Text="  Renaming existing tarball to %(PrevSBArchive.Identity).bak" Importance="High" />
+    <Move SourceFiles="%(PrevSBArchive.Identity)" DestinationFiles="%(PrevSBArchive.Identity).bak" />
```

Comment:
What's the scenario for keeping this around.  The primary reason I ask it because of its size.  I had even thought about using the `tar -u` option to update the tarball just so that there aren't multiple copies on disk and so you wouldn't have to extract it.

Reply:
I'm not sure there's a compelling reason to keep it around.  I was keeping it around so I could determine that it was doing the right thing.  I could probably just delete it here.  I don't think `tar -u` is what we want.  From what I can tell, it just appends to the end of an existing archive and doesn't replace.

Reply:
What about `-r`?

Reply:
The docs say that `-r` is append (not replace): https://man7.org/linux/man-pages/man1/tar.1.html

Worse, it doesn't work with compressed archives: 

```
$ tar -r -f packages/archive/Private.SourceBuilt.Artifacts.0.1.0-6.0.100-bootstrap*tar.gz -C ../some-prebuilts/ .
tar: Cannot update compressed archives
tar: Error is not recoverable: exiting now
```

Reply:
🤢

Reply:
I changed it to delete rather than backup.